demos/common/python/models/utils.py

```diff
@@ -42,6 +42,22 @@ def __init__(self, xmin, ymin, xmax, ymax, score, id, landmarks_x, landmarks_y):
             self.landmarks.append((x, y))
 
 
+class InputTransform:
+    def __init__(self, reverse_input_channels, mean_values, scale_values):
+        self.reverse_input_channels = reverse_input_channels
+        self.mean_values = np.array(mean_values, dtype=np.float32)
+        self.scale_values = np.array(scale_values, dtype=np.float32)
+
+    def is_trivial(self):
+        return np.array_equal(self.mean_values, [0., 0., 0.]) and \
+               np.array_equal(self.scale_values, [1., 1., 1.]) and \
+               not self.reverse_input_channels
+
+    def apply(self, inputs):
+        if self.reverse_input_channels:
```

Comment:
If the transform is trivial, you should immediately return `inputs` to avoid useless operations. You should also cache the result of the triviality check so that you don't compare the vectors every time. I suggest you just remove the is_trivial function and instead initialize `is_trivial` in the constructor.

Reply:
Done